packages/grid/x-data-grid/src/hooks/features/filter/gridFilterUtils.ts

```diff
@@ -234,20 +236,60 @@ export const buildAggregatedFilterItemsApplier = (
     return null;
   }
 
-  return (row, shouldApplyFilter) => {
```

Comment:
Can you share a benchmark compared to the `master` branch?
Curious to learn what is the benefit in this PR on its own 🙂

Reply:
Filtering 100,000 rows on my device:

| Case | Result |
| --- | --: |
| Without | `70.2 +- 3.8` |
| With | `62.6 +- 4.7` |

The improvement is consistenly around 11% for a realistic number of rows, but with a small improvement as N grows.

Reply:
Great, thanks!